proposals/0000-adding-support-for-swift.md

```diff
@@ -0,0 +1,111 @@
+- Rolling out a new version
+- Moving as much shared logic as possible to C++ and away from Obj-C, separating them as distinctly as possible while adding new features and capabilities and continuing to roll out new versions (**Users should not be affected, and ideally the Library maintainers should not be affected either**)
+- Beginning the transition from Obj-C to Swift from the tests in the internals (**No effect on users or library maintainers**)
+- Moving the rest of the modules one at a time from Obj-C to Swift (**Library maintainers may need to update, if they wish to take advantage of the newer APIs or if we break change an APIs signature, but ideally it should only be in the direction of making it easier to maintain libraries and better APIs**)
+- Updating CocoaPods infra
+- ? (Please add any more steps that come to mind, at the end or anywhere else in the list)
+
+## How we teach this
+
+This should be a pretty welcome step for the user base (and even the maintainers, if done correctly and without too many breaking changes). The docs for creating Native module will need to be updated to add support for Swift, and we can simply mention the change in React Native EU or similar conferences.
+
+The idea is to make it very un-noticeable to the vast majority of the user base who simply want to use the React side for creating their mobile applications, and the part of the community that is capable of creating native modules should simply feel like it is just another native Apple app that they can extend using whatever it is that they need (perhaps adding a watchOS target support for their app, or using CoreML or AppClips or making a SwiftUI based widget)
+
+It should be presented as React Native keeping up with changes and lowering the barrier of entry for all devs and that we are simply making part of the work that is duplicated across platforms easy to reuse, using the same tools that our industry is using and without sacrificing the experience.
+
+## Unresolved questions
+
+- What other parts of the infra need to be changed for this transition to work?
+- How will this affect other targets such as React Native macOS, React Native visionOS, etc.?
+- Is our current separation of internals from the OSS template enough that we can simply change the template directly without any of the first four steps needed? (I managed to get a [test repo](https://github.com/TheRogue76/React-Native-Template-With-Swift) to work as a proof of concept with these changes, but I am not sure if that means that the app will work for all situations (new arch, bridge-less, etc.). I also submitted [a PR](https://github.com/facebook/react-native/pull/41896) to test out the current tests we had and those passed, but that might not be enough). And what about Expo? Will it continue to function if this change is done to the base template, or would it interfere with any of the work done on their side
```

Comment:
> Is our current separation of internals from the OSS template enough that we can simply change the template directly without any of the first four steps needed?

No, as explained in the PR, that's not enough.
We had to make sure that all the configurations work and that the flags are passed properly. 

Also, the AppDelegate is not the only API that users interface with. There are all the APIs related to create a Custom Module and a Custom Component.

Reply:
Removed under [6a05439](https://github.com/react-native-community/discussions-and-proposals/pull/747/commits/6a0543926e0f59ec507fe97e4cd97834aaa7efde).